# Sample configuration file for the LTM server.
#
# It loads all plugins defined in this package
# - EmotionPlugin
# - LocationPlugin
# - ImageStreamPlugin
# - PeopleEntityPlugin, ObjectsEntityPlugin, RobotEntityPlugin, LocationEntityPlugin


# Server parameters
db:          "ltm_db"
collection:  "episodes"
host:        "localhost"
port:         27017
timeout:      60.0


# LTM plugins and parameters.
# Each plugin must define the pluginlib class and its parameters
plugins:

  # Emotion Engine pluginsrc/plugins_manager.cpp
  emotion:
    # Class to load through pluginlib
    class: "ltm_samples::EmotionPlugin"

  # Robot location plugin
  location:
    # Class to load through pluginlib
    class: "ltm_samples::LocationPlugin"

  # Stream plugins
  streams:
    # List of names to consider. 
    # For each name, you must declare its class and ROS parameters.
    include: ["image"]

    # ImageStream
    image:
      class: "ltm_addons::ImageStreamPlugin"
      type: "images"
      collection: "images"
      topic: "/robot/fake/sensors/camera/image_raw"
      buffer_frequency: 3.0
      buffer_size: 100

  # Entity plugins
  entities:
    # List of names to consider. 
    # For each name, you must declare its class and ROS parameters.
    # include: ["people", "objects", "robot", "location"]
    include: ["people"]

    # PeopleEntityPlugin
    people:
      class: "ltm_samples::PeopleEntityPlugin"
      type: "people"
      collection: "people"
      topic: "/robot/fake_short_term_memory/person/updates"


    # ObjectsEntityPlugin
    objects:
      class: "ltm_samples::ObjectsEntityPlugin"
      type: "objects"
      collection: "objects"
      topic: "/robot/fake_short_term_memory/object/updates"

    # RobotEntityPlugin
    robot:
      class: "ltm_samples::RobotEntityPlugin"
      type: "robot"
      collection: "robots"

    # LocationEntityPlugin
    location:
      class: "ltm_samples::LocationEntityPlugin"
      type: "location"
      collection: "locations"

